tests/common/widgets/chat/index.ts

```diff
@@ -2,26 +2,55 @@ import { fakeAct, skippableDescribe } from '../../common';
 
 import { createOptionsTests } from './options';
 
-import type { TestOptions, TestSetup } from '../../common';
+import type { SupportedFlavor, TestOptions, TestSetup } from '../../common';
 import type { ChatConnectorParams } from 'instantsearch.js/es/connectors/chat/connectChat';
 import type { ChatWidget } from 'instantsearch.js/es/widgets/chat/chat';
+import type { ChatProps } from 'react-instantsearch';
 
-type WidgetParams = Parameters<ChatWidget>[0];
-export type ChatWidgetSetup = TestSetup<{
-  // Explicitly adding ChatConnectorParams back. For some reason
-  // ChatConnectorParams are not inferred when Omit is used with WidgetParams.
-  widgetParams: Omit<WidgetParams, 'container'> & ChatConnectorParams;
+type JSBaseWidgetParams = Parameters<ChatWidget>[0];
+// Explicitly adding ChatConnectorParams back. For some reason
+// ChatConnectorParams are not inferred when Omit is used with WidgetParams.
+export type JSChatWidgetParams = Omit<JSBaseWidgetParams, 'container'> &
+  ChatConnectorParams;
+export type ReactChatWidgetParams = ChatProps<unknown>;
+
+type ChatWidgetParams<T extends SupportedFlavor = 'javascript'> = {
+  javascript: JSChatWidgetParams;
+  react: ReactChatWidgetParams;
+  vue: Record<string, never>;
+}[T];
+
+export type ChatWidgetSetup<T extends SupportedFlavor> = TestSetup<{
+  widgetParams: ChatWidgetParams<T>;
 }>;
 
-export function createChatWidgetTests(
-  setup: ChatWidgetSetup,
-  { act = fakeAct, skippedTests = {} }: TestOptions = {}
+declare module '../../common' {
+  interface FlavoredWidgetParams {
+    createChatWidgetTests: {
+      javascript: JSChatWidgetParams;
+      react: ReactChatWidgetParams;
+      vue: Record<string, never>;
+    };
```

Comment:
Would you be able to rely on an exported type based off of ChatWidgetSetup (before being scoped to `T`), instead of repeating the structure?

Reply:
Not sure I follow, to me it looks like this would be possible:

```ts
export type _FlavoredChatWidgetParams = {
  javascript: JSChatWidgetParams;
  react: ReactChatWidgetParams;
  vue: Record<string, never>;
};

export type ChatWidgetParams<T ...> = _FlavoredChatWidgetParams[T];

// ...

declare module '../../common' {
  interface FlavoredWidgetParams {
    createChatWidgetTests: _FlavoredChatWidgetParams
  }
}
```

All in all, this is a nitpick, so feel free to disregard.